tests/EmbeddedRelationsTest.php

```diff
@@ -951,4 +938,36 @@ public function testGetQueueableRelationsEmbedsOne()
         $this->assertEquals(['father'], $user->getQueueableRelations());
         $this->assertEquals([], $user->father->getQueueableRelations());
     }
+
+    public function testUnsetPropertyOnEmbed()
+    {
+        $user = User::create(['name' => 'John Doe']);
+        $user->addresses()->save(new Address(['city' => 'New York']));
+        $user->addresses()->save(new Address(['city' => 'Tokyo']));
+
+        // Set property
+        $user->addresses->first()->city = 'Paris';
+        $user->addresses->first()->save();
+
+        $user = User::where('name', 'John Doe')->first();
+        $this->assertSame('Paris', $user->addresses->get(0)->city);
+        $this->assertSame('Tokyo', $user->addresses->get(1)->city);
+
+        // Unset property
+        unset($user->addresses->first()->city);
+        $user->addresses->first()->save();
+
+        $user = User::where('name', 'John Doe')->first();
+        $this->assertSame(null, $user->addresses->get(0)->city);
+        $this->assertSame('Tokyo', $user->addresses->get(1)->city);
+
+        // Unset and reset property
+        unset($user->addresses->get(1)->city);
+        $user->addresses->get(1)->city = 'Kyoto';
+        $user->addresses->get(1)->save();
+
+        $user = User::where('name', 'John Doe')->first();
+        $this->assertSame(null, $user->addresses->get(0)->city);
```

Comment:
```suggestion
        $this->assertNull($user->addresses->get(0)->city);
```

Reply:
I've done this to be consistent with the other asserts, but it's just a very personal perspective.